### ISSUE TITLE
1.1 Game rebalancing

### ISSUE THREAD

**Issue by Bertram25** [ValyriaTear/ValyriaTear#391]
Prerequisites: ~~#107 #348 #385~~ (Quite a lot already!)

Once all that is done, balancing can be redone to make early enemies easy to win, and make the forest much more friendly in the first maps.
Then, snakes and bats will be slightly harder, until you reach the fenrir.

Once this is done, the mt elbrus will have to follow the same logic:
- [x] Add a way (later: escape smoke, now escape circles) to go back to a "safe" place when in normal wandering area.
- [x] Add custom attack animation support for enemies.
- [x] Ensure every std enemy has a strength and a weakness.
- [x] Ensure every enemy party is giving away something interesting. I mean besides a potion and moon juice.
- [x] Ensure the player can pay himself something using the material throughout the game so that characters always have something to improve.
  - [x] Adapt beginning shop, but only provide a few items, the characters can't stay long.
  - [x] Add a middle of dungeon shop with same items to permit buys and introduce the shop mushroom.
  - [x] Move the shop mushroom in a more visible location in the fleeing to mountain "dungeon".
  - [x] Remove Sophia in favor of the mushroom merchant and tweak content to be able to buy few interesting items.
  - [x] Add mushroom merchant at the temple exteriors in upper stairs to ease the fights.
  - [x] Add save point in upper stairs exteriors as well to ease when characters slips have to go through 3/4 difficult maps at once.
- [x] Ensure every standard enemy has a "special" attack; to have more variety. And review stats.
  - [x] Ratto (Won't do, very first simple enemy.)
  - [x] Slime (Particle effect from enemy on water spray)
  - [x] Spider (Animation on spider web)
  - [x] Fenrir 1st battle
  - [x] Snake (Better visuals for special attacks only)
  - [x] Slime mother (slightly better visuals only)
  - [x] Bats (Better visuals only for HP Drain)
  - [x] Fenrir 2nd battle
  - [x] Fenrir last battle
  - [x] Shroom (Better visuals for HP Drain and poison only)
  - [x] Eyeball (Change HP Drain to Magical def decrease, Better visuals for it)
  - [x] Beetle (Better visuals for stamina stealing)
  - [x] Rat (Better visuals for poison attack)
  - [x] Thing (Fire attack part of magic improvement.)
  - [x] Yeti (Won't do, only here to bash)
  - [x] Skeleton (Better visuals for frenzy attack)
- [x] Add a bit more variety in the different Harlequin's boss, to show the progression in them.
  - [x] Harlequin first battle should be untouched.
  - [x] Harlequin second and third battle: Uses all special attacks.
  - [x] Harlequin last battle, has a lot of phys def until all eyeballs are dead. (Magic shield animation missing)
  - [x] Make Harlequins flee the character but not very fast so that he is slightly slower than a walking character. Harlequins still randomly respawn away from the character.
- [x] Improve the Dorver boss as it is too easy for now.
  - [x] ~~The Dorver boss is physical. Let's use a counter before a party attack. E.g: 5, 4, ...~~
  - [x] It should be relatively hard but not that much.
- [x] Improve Andromalius to be a bit more twisted (while still based on magic)

**Comment by eugeneloza** [comment]
Came up with an idea on 'armormelt' potion. An expensive or artifact-like (weak potions are drop-only, but powerful one only in a hidden location chest) can save you against a heavy-armored target, like the black soldier or Dorover (maybe I misspelled it again?).
Maybe also some turn-undeads scrolls for few first skeletons fights (very expensive, but effective). And fire resistance orbs against 'the Thing's fire attacks in the Shrine.

**Comment by eugeneloza** [comment]
Another thought... Maybe first aid should be not that 'quick-to-get' (or much weaker / more expensive than now) to make the player use potions.

**Comment by Bertram25** [comment]
All valid points.
I don't know yet how much the armor-melt potion/scroll is feasible, but I'll definitely have a look at reducing money, and the new skills system will make skills improvable, and thus weak as long as you don't put enough points in them.

**Comment by eugeneloza** [comment]
I was thinking about grinding (you never know when you have to stop grinding and advance) and came up with idea that it may be avoided this 2 ways: XP for enemy is divided by 'average' party members levels (Arcuz way) or new levels take much more XP to level up but new toughter monsters provide respective increase (Old-school RPG way). That'll make inefficient grinding e.g. at lvl1 slimes and will force the player to move on at 'intended' difficulty (and maybe even force to constant and wise use of potions).

**Comment by akien-mga** [comment]
Both solutions could be interesting yes. I had also proposed the old school solution to make XP requirements per level and XP gain per monster level follow an (attenuated) exponential rule. So you'd be able to grind e.g. up to level 4 by fighting level 2 monsters, but it would soon become really tedious to gain more levels as your XP requirements would already be too high. That makes it relatively easy to balance the game as you can more or less expect that the players will always have (monster level ±1).

**Comment by Bertram25** [comment]
Used a new balance based on 'material' items which are to be exchanged either for items, or money.
Which already makes the game more interesting and gives interest in getting potions.

I also tweaked slimes and spiders stats and appearances to make the game beginning less steep.
Last but not least, I improved their base attacks visual and added special attacks to them.

**Comment by Bertram25** [comment]
Next step is to tweak the Fenrir, and the snakes, adding repeatables tutorials along the way.

**Comment by eugeneloza** [comment]
Great! I'm really looking forward to game continuation :)

**Comment by FulcrumA** [comment]
Random player who got the game off somewhere in 1.0 versiona nd decided to give it a try. While I am not involved with this project much, I did spend a few hours so I feel that I owe some feedback regarding balancing at the very least, even if it may be somewhat of a lengthy wall of text regarding mostly pacing:

Battles indeed need balancing. A thing prevalent from the start of the game is that it's horribly grindy. I understand that many JRPGs are based off that but here, especially in the beginning is just F, F, F, F, F, for several minutes, every minute or so. Grinding is necessary to get the levels and get anywhere but it gets really boring when everything if forcing one to F, F, F, F, F (as mana/sp/whatever you call it runs out very quickly and on abilities that often have miniscule impact).

It gets even worse later in the game. The ancients temple simply made me lose will to play anymore. Mana was necessary for healing, Brennon's special skills didn't make much of a difference, it was more F, F, F, F, F. At the chest with the rats I had to turn off the game, not because it was particularly hard, just because it was minutes, upon minutes, upon minutes of F, F, F, F, heal, F, F, F, F.

I don't need enemies where every single one of them has a special attack, status effect etc. Just makes fights faster - decreasing enemy health, even if it also means decreasing party members health gain to balance it out (though I'd recommend some alternative as it may force one to grind even more to make up for that) would help drastically, especially during those fights where party members have nothign to do but attack, attack, attack - and frankly, since anything but basic attack uses mana, sooner or later with how prolonged the fights are - they will have to. It could also be coupled with getting better gear - to be honest, getting an iron sword seemed to make no difference for Brennon when I started using it instead of the wooden one - enemies scaled enough that it was still wading through repetitive F, F, F, F, just now with new enemies.

Harlequin, for example, may have seemed interesting but in truth it was F, F, F, F, F since quickly in the round a mana debuff made one leak it fast enough that using potions to regain it was a waste. So, cue 10 - 15 minutes of F, F, F, F, F - first Harlequin clone, then some eyeballs, then another Harlequin clone, then some more eyeballs, over and over and over again. 

I had half a mind of getting some application to just keep on pressing F for me and leave the game in the background. Maybe if I'd feel ambitious enough I'd also consider making it try to heal characters' health when its low enough but that's beside the point - the point is that one shouldn't require some bot or trainer for a single player game and one which would be used only because the fights themselves are prolonged slog of dozens upon dozens of rounds. Funnily enough it's also why I grinded just pressing F, without looking at the screen in the game's beginning - to rack up enough levels to be able to get through fights later without having to stare upon dozens of rounds of exchanging basic blows with enemies every minute or so.

In light of that I understand the comment of handling grinding in that players are forced to progress rather than camp one area and keep on grinding but frankly, without that, I don't think I'd have strength and patience to get through the forest with how long and how bothersome each fight would be. And it's not the matter of it being jRPG at that point either - I didn't have such an issue playing any (including the very first installments of) Final Fantasy , Tales Of or Breath of Fire.

In kinda-sorta conclusion - balancing of the game won't be achieved through unique skills, loot or variety of mobs in enemy group as much as by making it possible to go through without forcing one to camp an area to grind - be it to be able to face the enemies or just go through the fights quick enough - and by making the fights shorter, more dynamic, with character skills having bigger impact and by enemies going down faster, as tedium is a greater than the enemies themselves.

**Comment by rootslinux** [comment]
Hi, stumbled upon your comment tonight and just wanted to give you my personal response. i agree that it's grindy. I have a special distaste for any game where you spam the "attack" command until the enemy is dead. It's what I've desperately tried to avoid in my own game's battle system (which is the grandparent of ValyriaTear).

That said, its very easy to say "this game needs to be better balanced". The truth is that good balance and polish takes a lot of time, effort, and can be rather tedious work. AFAIK, there was no person dedicated to the design and balance on VT. Bertram had to play equal parts developer, writer, designer, and so on. Speaking from personal experience, it's very difficult to balance all of those roles, so I can't fault him for having a messy game balance. I think the intention was to do iterative improvements on this 1.0 release, and for a while it looked like that may happen, but Bertram probably burnt out after hitting this huge achievement and lacked the will, energy, and possibly time to keep this project alive. He busted his ass for a good couple of years to make this release happen, and you have to respect all that he was able to achieve in that time, even considering that he got a great headstart thanks to Hero of Allacrost.

If this project were still active, I'd suggest you lend a hand and do some of the balancing work yourself. That's what open source game projects should be about anyway; everyone jumping in to help improve the games that they are interested in.

**Comment by FulcrumA** [comment]
>Hi, stumbled upon your comment tonight and just wanted to give you my personal response.

Oh, thank you. That's kind of you, especially since I didn't expect anything like this, given that - as mentioned - the project seems to be kinda dead by now and I felt merely obliged to leave my own impressions after playing the game, owing it to a person directing me to it.

>I can't fault him for having a messy game balance. I think the intention was to do iterative improvements on this 1.0 release, and for a while it looked like that may happen

I don't really fault the person, merely complain about the aspect of the game. I am aware that from developer's side the balance may look different than when one is grinding through the whole thing.

>He busted his ass for a good couple of years to make this release happen, and you have to respect all that he was able to achieve in that time, even considering that he got a great headstart thanks to Hero of Allacrost.

I do respect the will to do something and doing something - the game is short (well, the episode available is) but for a simple story and grindy battle, it had fun moments reminiscing of the oldschool jRPGs and those I appreciate.

>If this project were still active, I'd suggest you lend a hand and do some of the balancing work yourself.

Hah, I doubt I'd be able to modify the code I am unfamiliar with, likely not in the language I know, but otherwise, assuming other projects I spend my time on already and the daily life would still leave me some free time, I wouldn't mind to look to contribute in some at least token way if this thing would be still otherwise developed.

>That's what open source game projects should be about anyway; everyone jumping in to help improve the games that they are interested in.

Aye.

**Comment by rootslinux** [comment]
I think you'd be surprised at how easy it is to do balance work. Unless you are making major design/feature changes, the characters and enemy stats are pretty much in plain text and simple to modify. The problem with VT really is that there's no wiki or any documentation really; just the github page and the blogspot for news updates, so the barrier for entry is higher than it should be.

I'm not sure exactly where VT stores its stats and other balancing data (I did some work for this project several years ago after it forked from my own). But for my own project, here are links to Lua files that you can glance through and as you can see, it's pretty intuitive and easy to change.

- https://bitbucket.org/allacrost/allacrost/src/default/lua/data/actors/characters.lua
- https://bitbucket.org/allacrost/allacrost/src/default/lua/data/actors/enemies_set_01.lua
- https://bitbucket.org/allacrost/allacrost/src/default/lua/data/inventory/weapons.lua
- https://bitbucket.org/allacrost/allacrost/src/default/lua/data/skills/attack.lua

We also have some documentation for designers, although not all of these pages are fully fleshed out.
http://allacrost.org/wiki/index.php?title=Game_Designer_Main


If you're interested in these sort of old school RPGs, keep an eye on Hero of Allacrost. It's still an active project, and we're close to publishing a new release soon. I'd sure appreciate feedback like the type you've given here when we publish our release candidates, so that we can make sure we don't give players the same frustrating experience with balance that you shared here in the final release. Thanks!

**Comment by FulcrumA** [comment]
@rootslinux 
I'm reluctant to make any real promises but can t least say that I will strive to remember to check on the project you mention once in a while and will give it at least a try once the new version you mention will come out :)

**Comment by Bertram25** [comment]
@FulcrumA Hey there, :)

Sorry to make you think the project is dead, my latest commit was actually made on the Sep 17, 2018, and I've got a couple more pending that I need to cleanup before push, even though, this is refactoring.

I'll try to answer each interesting points here as much as I can and I must say I do appreciate your feed-back and I believe most of it is true.

First of all, let's clean the dust on a few points, nope, the project is not dead and there have been quite a lot of bug squashed and features added or improved since 1.0. Let's the figures talk: 904 commits since 1.0, and 100 issues closed.

So why hasn't the next release happened? Because I got caught in a few traps. Number one is, I wanted to add what happened to be bigger features than I thought, mostly to name it, the skill tree feature that permits to improve characters stats and obtain skill a bit alla Final Fantasy X.

To release 1.1 (which would deserve a 2.0 when I think about the time spent ;]), I have 5 issues left and guess what?, the game balancing and boss improvement is part of those issues.
On the master branch, I've rebalanced up to the Harlequin boss, and I now need to make more fun. But I also wanted to kill some cleanups and refactorings I procrastinate since years.

Let's also be honest, I've been caught with life a lot, and I have preferred my family life over coding, for obvious reason. So, indeed I cannot spend as much time as before on it, and progress are slow, but on the other hand, my motivation has increased a lot now I've got a rather good balance between code and real life. ;)

1.1 should happen "soon", but that "soon" could be another 3 to 6 months depending on the time and help I have. At which point, I'll have a refined, a bit more fun 1.0 story and will be able to move on with the story with a somewhat more scalable code.

Now let's go to the meat: grinding and game balancing. I agree with everything but here is a more elaborated answer:

- ... "by making the fights shorter",
Check, I did it. I agree the balancing at the beginning was making the game tedious. battle are way shorter in the master branch if I didn't break something stupidly (Around 10s each max on average).
And the enemies are now "mapped" (by hand for now) to make the game progression (way) faster.
As an example, you can now go until half the forest at level 3 and the skill tree make you able to progress your stat almost on each battle, making you improve on the way. 

That said, as soon as I can have another one review on this, fixes will happen.

-  more dynamic:
On this, I still believe that monsters should have special skills. In special skills, I don't mean that the first one encountered should throw magic meteorites with blazing flames at you, but they will try to eat a little of your stamina, just to have some effect on your characters and introduce effects.

Skills should have a higher impact, while it's still an area of work, and I do want them to have 2 kinds of effects at once, so you can use a skill to E.g: deal a damage burst and apply a specific effect. This way, you can also use the skill to apply the effect only as a strategy, ...

But remember, it's only the game beginning, and I wished the 2 characters to have around 3-4 skills at the end of episode I, not that more.

As for the mid-term future features:
As a mid-term (during episode I actually)
- I planned to make skills have levels (not in the farming way of things), but make them go bigger when you trade. Fake E.g.: Trade 'Cure' to 'Cure all'.

- Add the possibility for characters to go on one another skill paths. Actually, this is ready, I just need to find the right way and game time to put that in.

As for the dynamism of turned-based battle, I'd like to keep simple battles short but with somewhat nasty enemies as described above, but at the same time, keep longer boss battles. 
Turn based battles are boring by essence, so to add dynamism, you need to add enemies with strength and weaknesses, battle phases, agression level and all that kind of things.
So I'll contradict @rootslinux here, balancing is hellish and must be done with care.
Valyria is also using lua text file for enemies stats but this is a lot of work.

All in all, thanks again for your comment. :) It's cool to have some honest try of one's own open-source hobby. I hope to provide the community with some preview release soon(er) and in the meanwhile, if you can try compiling the master branch, I would love to hear from you on this one.

All the best,

**Comment by rootslinux** [comment]
Awesome, I'm glad to hear that VT is still under development. I'm really excited to see what the 1.1 release is like. It sounds like it is a refined/polished 1.0 release rather than a release having new playable content, is that right?

Also I think rather than saying that you were contradicting me, you meant to say "concur". :) If you've found a good system for doing balancing work, I'm eager to learn about it (maybe its good blog post material).

**Comment by Bertram25** [comment]
@rootslinux You're right. Atm, I'm refining gameplay and balancing as a part of the current content. Hence the 1.1 name and not the 2.0 one.

And as for the contradicting part, I meant to contradict this sentence: "I think you'd be surprised at how easy it is to do balance work"
To me, changing the numbers is indeed simple. Making them work together on the long term is as hard as making a good story. ;)

With the help and Pull requests from @gunchleoc , who knows, maybe I'll be able to release a Windows installer sooner than I thought? :)

**Comment by rootslinux** [comment]
Ah, now I see what you mean. Yes, changing balancing numbers is easy. But actually _balancing_ a game well is a lot of tough work and trial-and-error. That's what I meant to say by my earlier statement.

Over the last couple of years I've tried, unsuccessfully, to recruit someone who would focus on balancing and other design work that I frankly don't have time for right now. If memory serves correctly I think I might have helped out with some of the balancing on VT years ago and I recall it being really tedious and time-consuming.

**Comment by Bertram25** [comment]
Note to self, let's use #615 to improve the boss experience.

**Comment by Bertram25** [comment]
it's been a long time, but it's done
